src/Security/Authentication/Core/src/AuthenticationWebApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// Extension methods for setting up authentication services in an <see cref="WebApplicationBuilder" />.
+/// </summary>
+public static class AuthenticationWebApplicationBuilderExtensions
+{
+    /// <summary>
+    /// Registers services required by authentication services.
+    /// </summary>
+    /// <param name="builder">The <see cref="WebApplicationBuilder"/>.</param>
+    /// <returns>A <see cref="AuthenticationBuilder"/> that can be used to further configure authentication.</returns>
+    public static AuthenticationBuilder AddAuthentication(this WebApplicationBuilder builder)
+    {
+        if (builder == null)
+        {
+            throw new ArgumentNullException(nameof(builder));
+        }
+
+        builder.Services.AddAuthenticationCore();
+        builder.Services.AddDataProtection();
+        builder.Services.AddWebEncoders();
+        builder.Services.TryAddSingleton<ISystemClock, SystemClock>();
+        builder.Services.TryAddSingleton<IAuthenticationConfigurationProvider, DefaultAuthenticationConfigurationProvider>();
+        builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<AuthenticationOptions>, AuthenticationConfigureOptions>());
```

Comment:
Why doesn't this just call AddAuthentication? Why is it adding so many of these services manually? This should be equivalent to calling `builder.Services.AddAuthenticaiton().AddAuthorization()`. Anything we add to `WebApplicationBuilder` is sugar on top of the core APIs.